exporter/datadogexporter/internal/utils/api.go

```diff
@@ -27,17 +30,21 @@ func CreateClient(APIKey string, endpoint string) *datadog.Client {
 	return client
 }
 
+var ErrInvalidAPI = errors.New("API Key validation failed")
+
 // ValidateAPIKey checks that the provided client was given a correct API key.
-func ValidateAPIKey(logger *zap.Logger, client *datadog.Client) {
+func ValidateAPIKey(logger *zap.Logger, client *datadog.Client) error {
 	logger.Info("Validating API key.")
-	res, err := client.Validate()
-	if err != nil {
-		logger.Warn("Error while validating API key.", zap.Error(err))
-	}
-
-	if res {
+	valid, err := client.Validate()
+	if err == nil && valid {
 		logger.Info("API key validation successful.")
-	} else {
-		logger.Warn("API key validation failed.")
+		return nil
+	}
+	if err != nil {
+		msg := "Error while validating API key"
+		logger.Warn(msg, zap.Error(err))
+		return fmt.Errorf("%s: %w", msg, err)
```

Comment:
We should not return an error here: imagine there is an ongoing incident and the Datadog API returns a "500 Internal Server Error" error code; it could be that the API key is valid in this case and we just were not able to confirm that.

```suggestion
		return nil

Reply:
https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/9426/commits/006a462c613159e15b075ac7268936a471a40e01